app/javascript/shared/utils/JsTus.re

```diff
@@ -1,20 +1,19 @@
 [@bs.deriving abstract]
 type uploadOptions = {
   uploadUrl: string,
-  onError: (string) => string,
-  onSuccess: (string) => string,
+  onError: (string) => Js.nullable(string),
+  onSuccess: (unit) => Js.nullable(string)
 };
 
-[@bs.deriving abstract]
-type uploadObject = {
-  start: int
-};
-
-[@bs.module "tus-js-client"]
-external jsTusUpload: (Js.Json.t, uploadOptions) => uploadObject = "Upload";
+type uO;
+[@bs.module "tus-js-client"] [@bs.new]
+external jsTusUpload: (Js.Json.t, uploadOptions) => uO = "Upload";
```

Comment:
Setting the type of the _file_ as `Js.Json.t` won't work. `Js.Json.t` is the opaque type for JSON. That's not what a _file_ is.

The reality is that we don't care what the type of the file _is_. We only want to pass-through what we get from `event.files[0]` to this location. We can do that with an abstract type, coupled with a special identity function.

```reason
type file;

external makeFile: Js.t('a) => file = "%identity";
```

and then set the type of this argument as `file`.

```suggestion
external jsTusUpload: (file, uploadOptions) => uO = "Upload";
```

When passing in the file from outside, we can simply pass in any Js object `Js.t('a)` and have it converted to a `file` type with the special identity function.